pkgs/development/python-modules/plyer/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib, buildPythonPackage, fetchFromGitHub, mock, python3Packages, python, stdenv }:
+
+buildPythonPackage rec {
+
+  pname = "plyer";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "kivy";
+    repo = "plyer";
+    rev = version;
+    sha256 = "15z1wpq6s69s76r6akzgg340bpc21l2r1j8270gp7i1rpnffcjwm";
+  };
+
+  propagatedBuildInputs = [ python3Packages.keyring ];
+
+  checkInputs = [ mock ];
+  preCheck = ''
+    HOME=$(mktemp -d)
+    mkdir -p $HOME/.config/
+    touch $HOME/.config/user-dirs.dirs
+    mkdir -p $HOME/Pictures/
+  '';
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} -m unittest discover \
+      --start-directory plyer/tests \
+      --top-level-directory .
+    runHook postCheck
+  '';
+
+  postPatch = ''
+    # Fixup this file to
+    # - not assume ~/.config/user-dirs.dirs exists
+    # - not assume it actually has stuff in it if it does exist
+    # - return the path under HOME if user-dirs has no useful info.
+    # - use the right name for DOCUMENTS according to the spec
```

Comment:
Are there upstream issues for these bugs? If so it would be nice to include them. If not, we should probably report them.

Reply:
there is a pull request open for these

Reply:
We should link to those in the comment so that people reading in the future can have a paper trail

Reply:
Also if possible we should actually just patch with that pr using fetchpatch instead of manually doing it with a bunch of string replaces

Reply:
I submitted a PR and changed it to fetch the patches from that PR